mlir/lib/Dialect/Rock/Transforms/GridwiseGemmToBlockwise.cpp

```diff
@@ -810,42 +617,89 @@ struct GridwiseGemmRewritePattern : public OpRewritePattern<GridwiseGemmOp> {
+    Value storeBufferB = b.create<GpuAllocOp>(loc, loadBufferB.getType());
+
+    // We invert the transforms that are iter --> K x D slice of the tensor
+    // so that we can view loadBuffer as a K x D tensor
+    ArrayAttr loadBufferAViews =
+        invertTransforms(b, loc, maybeABufferViews->threadwiseView);
+    Value viewLoadBufferA = transform(b, loadBufferA, loadBufferAViews);
+    // Prior to LDS store, we need re-arrange register buffer to maxmize LDS
+    // vectorization Hence, creating the view w.r.t global that correspond to
+    // such re-arranged register buffer
+    auto [aKLdsStoreIterSplit, mLdsStoreiterSplit] =
+        createLDSStoreIterSplits(aCopyKPerThread, copyMPerThread, kpack);
+    FailureOr<GPUViews> maybeALdsStoreViews = createGemmInputTileViews(
+        b, loc, op.getA(), "m", bidGridOrder, bidGridLengths, gridSize,
+        blockSize, kPerBlock, mPerBlock, aKDimTidPartInfo, mDimTidPartInfo,
+        aKLdsStoreIterSplit, mLdsStoreiterSplit);
+    if (failed(maybeALdsStoreViews)) {
+      return failure();
+    }
+    ArrayAttr storeBufferAViews =
+        invertTransforms(b, loc, maybeALdsStoreViews->threadwiseView);
+    Value viewStoreBufferA = transform(b, storeBufferA, storeBufferAViews);
+    auto packALoop = packLoadBufferToStoreBuffer(
+        b, loc, elementTypeA, kpack, viewLoadBufferA, viewStoreBufferA);
+    ArrayAttr loadBufferBViews =
+        invertTransforms(b, loc, maybeBBufferViews->threadwiseView);
+    Value viewLoadBufferB = transform(b, loadBufferB, loadBufferBViews);
+    // Prior to LDS store, we need re-arrange register buffer to maxmize LDS
+    // vectorization Hence, creating the view w.r.t global that correspond to
+    // such re-arranged register buffer
+    auto [bKLdsStoreIterSplit, nLdsStoreIterSplit] =
+        createLDSStoreIterSplits(bCopyKPerThread, copyNPerThread, kpack);
+    FailureOr<GPUViews> maybeBLdsStoreViews = createGemmInputTileViews(
+        b, loc, op.getB(), "n", bidGridOrder, bidGridLengths, gridSize,
+        blockSize, kPerBlock, nPerBlock, bKDimTidPartInfo, nDimTidPartInfo,
+        bKLdsStoreIterSplit, nLdsStoreIterSplit);
+    if (failed(maybeBLdsStoreViews)) {
+      return failure();
+    }
+    ArrayAttr storeBufferBViews =
+        invertTransforms(b, loc, maybeBLdsStoreViews->threadwiseView);
+    Value viewStoreBufferB = transform(b, storeBufferB, storeBufferBViews);
+    auto packBLoop = packLoadBufferToStoreBuffer(
+        b, loc, elementTypeB, kpack, viewLoadBufferB, viewStoreBufferB);
+
     Type ldsReadTypeA = vectorTypeOrSelf(elementTypeA, kpack);
-    Type ldsReadTypeB = vectorTypeOrSelf(elementTypeB, kpack);
-    FailureOr<Value> maybeWrappedLdsA = wrapLDSBufferForStore(
-        b, loc, ldsBufferA, ldsReadTypeA, kpacksPerBlock, "m", mPerBlock,
-        aCopyKPerThread, copyMPerThread, aVectorDim);
+    FailureOr<Value> maybeWrappedLdsA =
+        wrapLDSBufferForStore(b, loc, ldsByteBufferA, ldsReadTypeA,
```

Comment:
Now wrapping here will produce K x D to flatLDS byte buffer because (tid, iter) to K x D is defined by the tile loading process , not here.

Reply:
or it could be define by a prior gemm for e.g.